src/combinatorics/simple_graph/ends/defs.lean

```diff
@@ -0,0 +1,232 @@
+connected_component_mk (G.induce Kᶜ) ⟨v, vK⟩
+
+/-- The set of vertices of `G` making up the connected component `C` -/
+@[reducible] def comp_out.supp (C : G.comp_out K) : set V :=
+{v : V | ∃ h : v ∈ Kᶜ, G.comp_out_mk h = C}
+
+@[ext] lemma comp_out.eq_iff_supp_eq (C D : G.comp_out K) : C = D ↔ C.supp = D.supp :=
+begin
+  split,
+  { rintro ⟨⟩, refl, },
+  { refine connected_component.ind₂ _ C D,
+    rintros ⟨v, hv⟩ ⟨w, hw⟩ h,
+    simp only [set.ext_iff, connected_component.eq, set.mem_compl_iff, set.mem_set_of_eq] at h ⊢,
+    exact ((h v).mp ⟨hv, reachable.refl _⟩).some_spec, }
+end
+
+instance : set_like (G.comp_out K) V :=
+{ coe := comp_out.supp,
+  coe_injective' := λ C D, (comp_out.eq_iff_supp_eq _ _).mpr, }
+
+lemma comp_out.mem_supp_iff {v : V} {C : comp_out G K} :
+  v ∈ C ↔ ∃ (vK : v ∈ Kᶜ), G.comp_out_mk vK = C := iff.rfl
+
+lemma comp_out_mk_mem (G : simple_graph V) {v : V} (vK : v ∈ Kᶜ) :
+  v ∈ G.comp_out_mk vK := ⟨vK, rfl⟩
+
+lemma comp_out_mk_eq_of_adj (G : simple_graph V) {v w : V} (vK : v ∈ Kᶜ) (wK : w ∈ Kᶜ)
+  (a : G.adj v w) : G.comp_out_mk vK = G.comp_out_mk wK :=
+by { rw [connected_component.eq], apply adj.reachable, exact a }
+
+namespace comp_out
+
+/--
+A `comp_out` specialization of `quot.lift`, where soundness has to be proved only for adjacent
+vertices.
+-/
+protected def lift {β : Sort*} (f : ∀ ⦃v⦄ (hv : v ∈ Kᶜ), β)
+  (h : ∀ ⦃v w⦄ (hv : v ∈ Kᶜ) (hw : w ∈ Kᶜ) (a : G.adj v w), f hv = f hw) : G.comp_out K → β :=
+connected_component.lift (λ vv, f vv.prop) $ (λ v w p, by
+  { induction p with _ u v w a q ih,
+    { rintro _, refl, },
+    { rintro h', exact (h u.prop v.prop a).trans (ih h'.of_cons), } })
+
+protected lemma ind {β : G.comp_out K → Prop} (f : ∀ ⦃v⦄ (hv : v ∈ Kᶜ), β (G.comp_out_mk hv)) :
+  ∀ (C : G.comp_out K), β C := by
+{ apply connected_component.ind, exact λ ⟨v, vnK⟩, f vnK, }
+
+/-- The induced graph on the vertices `C`. -/
+@[reducible]
+protected def coe_graph (C : comp_out G K) : simple_graph C.supp := G.induce (C : set V)
+
+lemma coe_inj {C D : G.comp_out K} : (C : set V) = (D : set V) ↔ C = D := set_like.coe_set_eq
+
+@[simp] protected lemma nonempty (C : G.comp_out K) : (C : set V).nonempty :=
+C.ind (λ v vnK, ⟨v, vnK, rfl⟩)
+
+protected lemma exists_eq_mk (C : G.comp_out K) : ∃ (v) (h : v ∈ Kᶜ), G.comp_out_mk h = C :=
+C.nonempty
+
+protected lemma disjoint_right (C : G.comp_out K) : disjoint K C :=
+begin
+  rw set.disjoint_iff,
+  exact λ v ⟨vK, vC⟩, vC.some vK,
+end
+
+lemma not_mem_of_mem {C : G.comp_out K} {c : V} (cC : c ∈ C) : c ∉ K :=
+λ cK, set.disjoint_iff.mp C.disjoint_right ⟨cK, cC⟩
+
+protected lemma pairwise_disjoint :
+  pairwise $ λ  C D : G.comp_out K, disjoint (C : set V) (D : set V) :=
+begin
+  rintro C D ne,
+  rw set.disjoint_iff,
+  exact λ u ⟨uC, uD⟩, ne (uC.some_spec.symm.trans uD.some_spec),
+end
+
+lemma eq_of_not_disjoint (C D : G.comp_out K) : ¬ disjoint (C : set V) (D : set V) → C = D :=
+comp_out.pairwise_disjoint.eq
```

Comment:
Not sure you need this lemma as it's a direct application of `set.pairwise_disjoint.eq`.